code/modules/mob/living/brain/brain_item.dm

```diff
@@ -579,6 +579,9 @@
 			qdel(actual_trauma)
 			return FALSE
 		actual_trauma.on_gain()
+		// certain traumas will qdel themselves in on_gain() if conditions aren't met
+		if(QDELETED(actual_trauma))
+			return FALSE
```

Comment:
would it be better for on_gain to return success/failure (in the style of like Initialize) to avoid deleting midway through setup, or is this sufficient (this is a genuine question)

Reply:
That would be cleaner and allow me to rewrite this code a bit better,  I'll rewrite the code to do that instead because on_gain is only called in two places